app/pods/publications/publication/translations/controller.js

```diff
@@ -0,0 +1,197 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { task } from 'ember-concurrency-decorators';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { PUBLICATION_EMAIL } from 'frontend-kaleidos/config/config';
+import CONSTANTS from 'frontend-kaleidos/config/constants';
+
+export default class PublicationsPublicationTranslationsController extends Controller {
+  @service store;
+  @service publicationService;
+
+  @tracked publicationFlow;
+  @tracked translationSubcase;
+  @tracked publicationSubcase;
+
+  @tracked showTranslationUploadModal = false;
+  @tracked showTranslationRequestModal = false;
+
+  get isRequestingDisabled() {
+    return this.translationSubcase.isFinished;
+  }
+
+  get isTranslationUploadDisabled() {
+    return this.model.length === 0;
+  }
+
+  @task
+  *saveTranslationUpload(translationUpload) {
+    const now = new Date();
+
+    // get latest Req activity
+    const requestActivity = this.model.filter((row) =>row.requestActivity !== null).sortBy('date').reverseObjects()[0].requestActivity;
+
+    const documentContainer = this.store.createRecord('document-container', {
+      created: now,
+    });
+    yield documentContainer.save();
+
+    const translationActivity = yield requestActivity.translationActivity;
+
+    // triggers call
+    const language = yield translationActivity.language;
+    const piece = this.store.createRecord('piece', {
+      created: now,
+      modified: now,
+      receivedDate: translationUpload.receivedAtDate,
+      confidential: false,
+      name: translationUpload.name,
+      file: translationUpload.file,
+      documentContainer: documentContainer,
+      language: language,
+      translationActivityGeneratedBy: translationActivity,
+    });
+    if (translationUpload.isSourceForProofPrint) {
+      piece.publicationSubcaseSourceFor = this.publicationSubcase;
+    }
```

Comment:
Niet meer relevant? De property komt op nog plaatsen voor in components, maar wordt nergens nog getoond in de UI. Moet dus opgekuist worden.